src/inmanta/ast/type.py

```diff
@@ -265,6 +262,34 @@ def type_string_internal(self) -> str:
         return self.type_string()
 
 
+@stable_api
+class Float(Number):
+    """
+    This class is an alias for the Number class and represents a float in
+    the configuration model.
+    """
+
+    def __init__(self) -> None:
+        Number.__init__(self)
+        self.try_cast_functions: Sequence[Callable[[Optional[object]], object]] = [float]
+
+    def validate(self, value: Optional[object]) -> bool:
+        """
+        Validate the given value to check if it satisfies the constraints
+        associated with this type
+        """
+        if isinstance(value, AnyType):
+            return True
+
+        if not isinstance(value, numbers.Number):
```

Comment:
Shouldn't this be `float`?

Reply:
I don't think so as `isinstance(1, float) == false` and then this doesn't work anymore:

```
entity A:
    float x
end
index A(x)
a = A(x=1)
```